File: frontend/check_brands.js
const { createClient } = require('@supabase/supabase-js');
const dotenv = require('dotenv');
const path = require('path');

dotenv.config({ path: path.resolve(__dirname, '.env') });

const supabase = createClient(
  process.env.SUPABASE_URL,
  process.env.SUPABASE_SERVICE_ROLE_KEY
);

async function check() {
  try {
    const { data: manufacturers, error: mError } = await supabase.from('manufacturers').select('*');
    if (mError) throw mError;

    console.log('--- MANUFACTURERS ---');
    manufacturers.forEach(m => console.log(`ID: ${m.id} | Name: ${m.name}`));

    for (const m of manufacturers) {
      console.log(`\nChecking: ${m.name}`);
      const { data: pricing, error: pError } = await supabase
        .from('manufacturer_pricing')
        .select('collection_name, door_style')
        .eq('manufacturer_id', m.id)
        .limit(10); // Sample test

      if (pError) {
        console.error(`Error for ${m.name}:`, pError.message);
        continue;
      }

      console.log(`Sample Row:`, pricing[0]);
      
      const { data: counts, error: cError } = await supabase
        .from('manufacturer_pricing')
        .select('collection_name', { count: 'exact', head: false })
        .eq('manufacturer_id', m.id);
      
      const uniqueCollections = [...new Set(counts?.map(c => c.collection_name))];
      console.log(`Total rows: ${counts?.length || 0}`);
      console.log(`Unique Collections:`, uniqueCollections);
    }
  } catch (err) {
    console.error('CRITICAL ERROR:', err.message);
  }
}

check();

File: backend/check_brands.py
import os
import asyncio
from dotenv import load_dotenv
from supabase import create_client, Client

# Load environment variables
load_dotenv()

supabase_url = os.environ.get("SUPABASE_URL")
supabase_key = os.environ.get("SUPABASE_SERVICE_ROLE_KEY")

if not supabase_url or not supabase_key:
    # Attempt to load from the .env file explicitly if not in environment
    print("Warning: SUPABASE_URL or SUPABASE_SERVICE_ROLE_KEY not found in environment. Checking local .env...")
    load_dotenv(os.path.join(os.path.dirname(__file__), '.env'))
    supabase_url = os.environ.get("SUPABASE_URL")
    supabase_key = os.environ.get("SUPABASE_SERVICE_ROLE_KEY")

if not supabase_url or not supabase_key:
    print("Error: SUPABASE_URL or SUPABASE_SERVICE_ROLE_KEY not found.")
    exit(1)

supabase: Client = create_client(supabase_url, supabase_key)

async def check():
    # 1. Get all manufacturers
    try:
        response = supabase.table('manufacturers').select('*').execute()
        manufacturers = response.data
    except Exception as e:
        print(f"Error fetching manufacturers: {e}")
        return

    print('--- MANUFACTURERS ---')
    for m in manufacturers:
        print(f"ID: {m['id']} | Name: {m['name']}")

    # 2. For each, check unique collections and door styles
    for m in manufacturers:
        print(f"\nChecking Manufacturer: {m['name']} ({m['id']})")
        
        try:
            # Fetch a sample of pricing data
            response = supabase.table('manufacturer_pricing') \
                .select('collection_name, door_style') \
                .eq('manufacturer_id', m['id']) \
                .limit(1000) \
                .execute()
            pricing = response.data
        except Exception as e:
            print(f"Error fetching pricing for {m['name']}: {e}")
            continue

        collections = set()
        styles = set()
        for p in pricing:
            if p.get('collection_name'):
                collections.add(p['collection_name'])
            if p.get('door_style'):
                styles.add(p['door_style'])

        print(f"Unique Collections (subset): {list(collections)[:10]}")
        print(f"Unique Styles (subset): {list(styles)[:10]}")
        print(f"Total Sample Size: {len(pricing)}")

if __name__ == "__main__":
    asyncio.run(check())
